src/bin/conserve/show.rs

```diff
@@ -0,0 +1,689 @@
+        l.push(format!("{:<20}", band_id));
+        let band = match Band::open(archive, &band_id) {
+            Ok(band) => band,
+            Err(e) => {
+                warn!("Failed to open band {:?}: {:?}", band_id, e);
+                continue;
+            }
+        };
+        let info = match band.get_info() {
+            Ok(info) => info,
+            Err(e) => {
+                warn!("Failed to read band tail {:?}: {:?}", band_id, e);
+                continue;
+            }
+        };
+
+        if options.start_time {
+            let start_time = info.start_time;
+            let start_time_str = if options.utc {
+                start_time.format(TIMESTAMP_FORMAT)
+            } else {
+                start_time
+                    .with_timezone(&chrono::Local)
+                    .format(TIMESTAMP_FORMAT)
+            };
+            l.push(format!("{:<10}", start_time_str));
+        }
+
+        if options.backup_duration {
+            let duration_str: Cow<str> = if info.is_closed {
+                info.end_time
+                    .and_then(|et| (et - info.start_time).to_std().ok())
+                    .map(duration_to_hms)
+                    .map(Cow::Owned)
+                    .unwrap_or(Cow::Borrowed("unknown"))
+            } else {
+                Cow::Borrowed("incomplete")
+            };
+            l.push(format!("{:>10}", duration_str));
+        }
+
+        if options.tree_size {
+            // TODO(MH): Readd a monitor here to indicate progress
+            let tree_mb_str = bytes_to_human_mb(
+                archive
+                    .open_stored_tree(BandSelectionPolicy::Specified(band_id.clone()))?
+                    .size(Exclude::nothing(), None)?
+                    .file_bytes,
+            );
+            l.push(format!("{:>14}", tree_mb_str,));
+        }
+
+        info!("{}", l.join(" "));
+    }
+    Ok(())
+}
+
+pub fn show_index_json(band: &Band) -> Result<()> {
+    // TODO: Maybe use https://docs.serde.rs/serde/ser/trait.Serializer.html#method.collect_seq.
+    let index_entries: Vec<IndexEntry> = band.index().iter_entries().collect();
+    let json = serde_json::to_string_pretty(&index_entries)
+        .map_err(|source| conserve::Error::SerializeIndex { source })?;
+    for line in json.lines() {
+        info!("{}", line);
+    }
+    Ok(())
+}
+
+pub fn show_entry_names<E: conserve::Entry, I: Iterator<Item = E>>(it: I) -> Result<()> {
+    for entry in it {
+        info!("{}", entry.apath());
+    }
+    Ok(())
+}
+
+pub fn show_diff<D: Iterator<Item = DiffEntry>>(diff: D) -> Result<()> {
+    // TODO: Consider whether the actual files have changed.
+    // TODO: Summarize diff.
+    // TODO: Optionally include unchanged files.
+    for de in diff {
+        info!("{}", de);
+    }
+
+    Ok(())
+}
+
+enum NutmegMonitorState<T: nutmeg::Model> {
+    View {
+        view: Arc<Mutex<View<T>>>,
+        _log_guard: ViewLogGuard,
+    },
+    Simple {
```

Comment:
`Simple` now `NoProgress` is used when the `no_progress` cli argument is passed.  
Piping the output trough nutmeg when terminal process is not enabled will at least require one more mutex each written line ([here](https://github.com/sourcefrog/conserve/pull/173/files#diff-c7640cf9576513aa91ce6fc75806bcfbdcc03a2de2ccb0b6f93e88f007267c5fR29)). I'm not sure if `nutmeg` itself syncs anything  as well, but I would suspect it does. This is just an unnecessary performance hit when logging.

Reply:
I'm kind of doubtful that any performance difference will be noticeable: 

Normally the lock will be uncontended, and an uncontended lock will probably just be an atomic swap and very cheap. Generally I would not complicate any code to avoid such a lock without evidence.

Normally not many lines are written to stdout, so even if contended it will not be hot. There are two cases where there might be a lot of output:

1. Verbose logging is on. If that goes to a terminal then painting the terminal will probably be the slow part. If it's going to a file perhaps not so much.
2. Bulk output like json, which probably should have some kind of lock for the whole large block, since nothing else should be interleaved with it.